homeassistant/components/sensor/__init__.py

```diff
@@ -365,7 +365,7 @@ def _is_valid_suggested_unit(self, suggested_unit_of_measurement: str) -> bool:
         unit converter supports both the native and the suggested units of measurement.
         """
         # Make sure we can convert the units
-        if (
+        if self.native_unit_of_measurement != suggested_unit_of_measurement and (
```

Comment:
```suggestion
        if self.native_unit_of_measurement == suggested_unit_of_measurement:
            return True
        if (
```

lets keep this in a separate early-return, since the condition is already quite long.